lib/service/modules/validation.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2016 Kinvey Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const util = require('util');
+
+const validationErrors = {
+  min: '%s must be greater than or equal to %d',
+  max: '%s must be less than or equal to %d',
+  pattern: '%s must match regex of %s',
+  required: '%s is required'
+};
+
+const validate = (expected, validationType, propertyName, propertyValue) => {
+  let valid = false;
+  let error = null;
+
+  switch (validationType) {
+    case 'min':
+      if (typeof propertyValue !== 'undefined' && propertyValue !== null) {
```

Comment:
Since every one of these checks for `propertyValue` existing, you can probably extract that out to a `propertyValidExists` boolean